lib/rules/no-insecure-url.js

```diff
@@ -88,15 +115,26 @@ module.exports = {
                     const rawStringText = node.value.raw;
                     const cookedStringText = node.value.cooked;
 
-                    if ((matches(blocklist, rawStringText) && !matches(exceptions, rawStringText)) ||
-                        (matches(blocklist, cookedStringText) && !matches(exceptions, cookedStringText))) {
-                        context.report({
-                            node: node,
-                            messageId: "doNotUseInsecureUrl"
-                        });
-                    }
-                }
-            }
+                  if (shouldFix(varExceptions,context, node) && (matches(blocklist, rawStringText) && !matches(exceptions, rawStringText)) ||
+                      (matches(blocklist, cookedStringText) && !matches(exceptions, cookedStringText))) {
+                      context.report({
+                        node: node,
+                        messageId: "doNotUseInsecureUrl",
+                        fix(fixer) {
+                          // Only fix if it contains an http url
+                          if (node.value.raw.toLowerCase().includes("http")) {

```

Comment:
If we had a blocklist that only contains "http" in our eslint config, this check wouldn't be necessary.
But then, we'd have to identify the specific disallowed protocol and find the correct replacement, which is trickier.

Let's leave it in for now. Just a thought.